build/downloader.ts

```diff
@@ -0,0 +1,114 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as releases from '@hashicorp/js-releases';
+
+function getPlatform(platform: string) {
+  if (platform === 'win32') {
+    return 'windows';
+  }
+  if (platform === 'sunos') {
+    return 'solaris';
+  }
+  return platform;
+}
+
+function getArch(arch: string) {
+  // platform | terraform-ls  | extension platform | vs code editor
+  //    --    |           --  |         --         | --
+  // macOS    | darwin_amd64  | darwin_x64         | ✅
+  // macOS    | darwin_arm64  | darwin_arm64       | ✅
+  // Linux    | linux_amd64   | linux_x64          | ✅
+  // Linux    | linux_arm     | linux_armhf        | ✅
+  // Linux    | linux_arm64   | linux_arm64        | ✅
+  // Windows  | windows_386   | win32_ia32         | ✅
+  // Windows  | windows_amd64 | win32_x64          | ✅
+  // Windows  | windows_arm64 | win32_arm64        | ✅
+  if (arch === 'ia32') {
+    return '386';
+  }
+  if (arch === 'x64') {
+    return 'amd64';
+  }
+  if (arch === 'armhf') {
+    return 'arm';
+  }
+  return arch;
+}
+
+interface ExtensionInfo {
+  extensionVersion: string;
+  languageServerVersion: string;
+}
+
+function getExtensionInfo(): ExtensionInfo {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const pjson = require('../package.json');
+  return {
+    extensionVersion: pjson.version,
+    languageServerVersion: pjson.langServer.version,
+  };
+}
+
+async function run(platform: string, architecture: string) {
+  const cwd = path.resolve(__dirname);
+
+  const buildDir = path.basename(cwd);
+  const repoDir = cwd.replace(buildDir, '');
+  const installPath = path.join(repoDir, 'bin');
+  if (fs.existsSync(installPath)) {
+    console.log('terraform-ls path exists. Exiting');
+    return;
+  }
+
+  fs.mkdirSync(installPath);
+
+  const extInfo = getExtensionInfo();
+  console.log(extInfo);
+
+  // userAgent = `Terraform-VSCode/${extensionVersion} VSCode/${vscodeVersion}`;
+  const ciBuild = process.env.CI;
+  const runnerLocation = ciBuild ? `CLI-Downloader GitHub-Actions` : `CLI-Downloader`;
+  const userAgent = `Terraform-VSCode/${extInfo.extensionVersion} ${runnerLocation}/(${platform}; ${architecture})`;
```

Comment:
```suggestion
  const userAgent = `Terraform-VSCode/${extInfo.extensionVersion} ${runnerLocation} (${platform}; ${architecture})`;
```
Version would typically follow `/` in User-Agent, and metadata is more often separated by space in braces.